src/coreclr/clrdefinitions.cmake

```diff
@@ -169,11 +169,7 @@ else()
   add_compile_definitions($<$<BOOL:$<TARGET_PROPERTY:CROSSGEN_COMPONENT>>:FEATURE_PREJIT>)
 endif(FEATURE_PREJIT)
 
-if(CLR_CMAKE_TARGET_WIN32 OR CLR_CMAKE_TARGET_LINUX)
-    add_compile_definitions($<$<AND:$<NOT:$<BOOL:$<TARGET_PROPERTY:CROSSGEN_COMPONENT>>>,$<NOT:$<BOOL:$<TARGET_PROPERTY:DAC_COMPONENT>>>>:FEATURE_PROFAPI_ATTACH_DETACH>)
-endif(CLR_CMAKE_TARGET_WIN32 OR CLR_CMAKE_TARGET_LINUX)
-
-add_compile_definitions($<$<BOOL:$<TARGET_PROPERTY:DAC_COMPONENT>>:DATA_PROFAPI_ATTACH_DETACH>)
+add_compile_definitions($<$<AND:$<NOT:$<BOOL:$<TARGET_PROPERTY:CROSSGEN_COMPONENT>>>,$<NOT:$<BOOL:$<TARGET_PROPERTY:DAC_COMPONENT>>>>:FEATURE_PROFAPI_ATTACH_DETACH>)
```

Comment:
@jkotas Could you explain why we don't enable attach/detach when `TARGET_PROPERTY:DAC_COMPONENT` is set?

Reply:
The [DAC](https://github.com/dotnet/runtime/blob/master/docs/design/coreclr/botr/dac-notes.md) build of the runtime does not need the code that does profiler attach. I guess that no defining `FEATURE_PROFAPI_ATTACH_DETACH` was the easiest way to make the DAC build work.

You can try defining `FEATURE_PROFAPI_ATTACH_DETACH` for DAC and see where the build breaks are. If it builds fine or if the build breaks are trivial to fix, it would be fine to define `FEATURE_PROFAPI_ATTACH_DETACH` unconditionally.

Reply:
Got it, thanks!